pylint/message/message_handler_mix_in.py

```diff
@@ -226,9 +232,15 @@ def is_one_message_enabled(self, msgid, line):
                 return self._msgs_state.get(msgid, fallback)
             return self._msgs_state.get(msgid, True)
 
-    def add_message(
-        self, msgid, line=None, node=None, args=None, confidence=None, col_offset=None
-    ):
+    def add_message(  # type: ignore # MessagesHandlesMixIn is always mixed with PyLinter
+        self: "PyLinter",
+        msgid: str,
```

Comment:
`msgid` is a special string with one letter between C W I F R, then 4 digits (C0111)  and `symbol` is a mix of alphabetic character and `-` (missing-docstring). A lot of function in `pylint.message` can takes both but not everything is named correctly right now. You can check `pylint.message.MessageIdStore` & co.


```suggestion
        msgid_or_symbol: MessageId, # MessageId already exists, it seems it would require a refactor or a new type
```

Reply:
You mean create a new type alias called `MessageId` with the specifics discussed in your comment?
I can try and do this, but I think this might create some issues by being too specific. As in, `mypy` will then want us to make sure that every `msgid_or_symbol` conforms to the rules of `msgid_or_symbol` and we might need to add checks all over the code base for this. Still, this is something to try!

Personally I would like to do the following sequence: this PR > `OutputLine` refactor + typing > `msgid_or_symbol` refactor. This will give me the typing of the first two PRs wil working on the `msgid_or_symbol` refactor.

Reply:
I don't know if mypy can check that a string type is a subset of str. I was thinking of `MessageId` as simply a semantic alias for `str` (The `MessageId` class is both str value (C0111 / missing-docstring) retrieved from the MessageStore using only one of the value and guaranteed to be coherent).

Reply:
> Not sure if this would actually provide a benefit.

I agree. There are other changes which are much more useful.
So probably nothing we should pursue in the immediate future.

Reply:
Yes the simple change to do is to keep the string type and rename args ``msgid`` to ``msgid_or_symbol`` (the convention I used in ``pylint.messages``)

Reply:
Couldn't that create problems with extensions explicitly using `msgid`?

Reply:
Yeap, probably a breaking change, let's put in 3.0 or not at all.

Reply:
IMO probably not worth doing. It could break too many things.